cmd/system-probe/api/module/loader.go

```diff
@@ -80,6 +83,15 @@ func Register(cfg *config.Config, httpMux *mux.Router, factories []Factory) erro
 		log.Infof("module %s started", factory.Name)
 	}
 
+	if !driver.IsNeeded() {
+		// if running, shut it down
+		log.Debug("system-probe module initialization complete, driver not needed, shutting down")
```

Comment:
nit: make it clear that the driver is the thing shutting down
```suggestion
		log.Debug("system-probe module initialization complete, driver not needed, shutting down driver")
```

Reply:
Good call